lib/processors/stringEscaper.js

```diff
@@ -0,0 +1,83 @@
+const escapeUnicode = require("escape-unicode");
+
+/**
+ * @see https://en.wikipedia.org/wiki/ASCII
+ * ascii contains 128 characters.
+ * its char codes reach from 0 to 127.
+ * @type {number}
+ */
+const CHAR_CODE_OF_LAST_ASCII_CHARACTER = 127;
+
+// use memoization for escapeUnicode function for performance
+const memoizeEscapeUnicodeMap = {};
+const memoizeEscapeUnicode = function(sChar) {
+	if (memoizeEscapeUnicodeMap[sChar]) {
+		return memoizeEscapeUnicodeMap[sChar];
+	}
+	memoizeEscapeUnicodeMap[sChar] = escapeUnicode(sChar);
+	return memoizeEscapeUnicodeMap[sChar];
+};
+
+/**
+ * Escapes non ASCII characters with unicode escape sequences.
+ *
+ * @see https://en.wikipedia.org/wiki/ASCII
+ * @see https://tools.ietf.org/html/rfc5137#section-6.1
+ *
+ *
+ * @param {string} string input string with non ascii characters, e.g. L♥VE
+ * @returns {{string: (string), modified: boolean}} output string with all non ascii
+ * characters being escaped by unicode sequence, e.g. L\u2665VE
+ */
+const escapeNonAscii = function(string) {
+	let result = "";
+	let modified = false;
+	for (let i = 0; i < string.length; i++) {
+		const char = string[i];
+		// check for non ascii characters (characters which have a char code
+		// greater than the ascii character code range)
+		if (string.charCodeAt(i) > CHAR_CODE_OF_LAST_ASCII_CHARACTER) {
+			result += memoizeEscapeUnicode(char);
+			modified = true;
+		} else {
+			result += char;
+		}
+	}
+	return {
+		modified,
+		string: result
+	};
+};
+
+/**
+ * Escapes non ASCII characters with unicode escape sequences.
+ *
+ * @public
+ * @alias module:@ui5/builder.processors.stringEscaper
+ * @param {Object} parameters Parameters
+ * @param {module:@ui5/fs.Resource[]} parameters.resources List of resources to be processed
+ * @param {Object} [parameters.options] Options
+ * @param {string} [parameters.options.encoding="utf8"] resource file encoding based on supported
```

Comment:
We need to agree on a place to make the transformation between ui5.yaml configuration (`UTF-8`/`ISO-8859-1`) and the Node.js encodings strings (`utf8`/`latin1`).

Currently I think I can put `"UTF-8"` in my ui5.yaml and it will be passed to `.toString` here which only understands `"utf8"`.

Reply:
the only "unsupported" one is: ISO-8859-1.
https://encoding.spec.whatwg.org/#names-and-labels
UTF8 / utf8 / UTF-8 are all supported by node

Reply:
https://encoding.spec.whatwg.org/#names-and-labels is not the correct resource for what is supported by Node.js. It is referred to as being what "Modern Web browsers follow", not Node.js.

I just tested the following to confirm this:
```
> Buffer.from("🐬", "unicode-1-1-utf-8").toString()
Thrown:
TypeError [ERR_UNKNOWN_ENCODING]: Unknown encoding: unicode-1-1-utf-8
```

The [Node.js documentation](https://nodejs.org/api/buffer.html#buffer_buffers_and_character_encodings) states a list of strings that is supported. `"UTF-8"` seems to work, but is not documented and supposedly only supported for legacy reasons. There are even shortcuts for the documented `utf8` in the code: https://github.com/nodejs/node/blob/master/lib/internal/util.js#L104

I would prefer an explicit mapping to what is stated in the Node.js API docs just to be safe.

Reply:
As discussed with @tobiasso85:

Make the `parameters.options.encoding` input parameter of the **task** reflect the ui5.yaml configuration options. So either `"UTF-8"` or `"ISO-8859-1"`. Make the parameter mandatory with no default (there is no point in using the task if the encoding is unknown).

The **processors** `parameters.options.encoding` input parameter should accept all Node.js supported encodings as listed at https://nodejs.org/api/buffer.html#buffer_buffers_and_character_encodings. There should not be any additional mapping (like `ISO-8859-1` to `latin1`) in the processor.

Reply:
indeed